pkg/n2k/sequence.go

```diff
@@ -0,0 +1,60 @@
+package n2k
+
+import (
+	"fmt"
+	"time"
+)
+
+// Calculated with math, but reference:
+// https://copperhilltech.com/blog/what-is-the-difference-between-sae-j1939-and-nmea-2000/
+const (
+	//	maxBytesInFastPacket = 223
+	MaxFrameNum = 31
+)
+
+type sequence struct {
+	started  time.Time
+	zero     *packet // packet 0 of sequence
+	expected uint8
+	received uint8
+	contents [MaxFrameNum + 1][]uint8 // need arrays since packets can be received out of order
+}
+
+func (s *sequence) add(p *packet) {
+	if p.frameNum == 0 {
```

Comment:
should we check in here if the frame specified has already been received and processed, and error the whole packet/reset?

Reply:
I now reset the sequence when detecting a conflict and log a warning. The conflict is same seqid and framenum from the same source, so we must have missed packets on the previous send, so we should start over. Somewhere I saw mention of a time limit for multiple frame transmission--when I find it again I'll add code to prune orphaned sequences.